exec/java-exec/src/test/java/org/apache/drill/test/QueryTestUtil.java

```diff
@@ -100,36 +96,125 @@ public static String normalizeQuery(final String query) {
   }
 
   /**
-   * Execute a SQL query, and print the results.
+   * Execute a SQL query, and output the results.
    *
    * @param drillClient drill client to use
    * @param type type of the query
    * @param queryString query string
+   * @param print True to output results to stdout. False to log results.
+   *
    * @return number of rows returned
    * @throws Exception
    */
-  public static int testRunAndPrint(
-      final DrillClient drillClient, final QueryType type, final String queryString) throws Exception {
+  private static int testRunAndOutput(final DrillClient drillClient,
+                                      final QueryType type,
+                                      final String queryString,
+                                      final boolean print) throws Exception {
     final String query = normalizeQuery(queryString);
     DrillConfig config = drillClient.getConfig();
     AwaitableUserResultsListener resultListener =
-        new AwaitableUserResultsListener(
-            config.getBoolean(TEST_QUERY_PRINTING_SILENT) ?
-                new SilentListener() :
-                new PrintingResultsListener(config, Format.TSV, VectorUtil.DEFAULT_COLUMN_WIDTH)
-        );
+      new AwaitableUserResultsListener(print ?
+      new PrintingResultsListener(config, Format.TSV, VectorUtil.DEFAULT_COLUMN_WIDTH):
+      new LoggingResultsListener(config, Format.TSV, VectorUtil.DEFAULT_COLUMN_WIDTH));
     drillClient.runQuery(type, query, resultListener);
     return resultListener.await();
   }
 
+  /**
+   * Execute one or more queries separated by semicolons, and output the results.
+   *
+   * @param drillClient drill client to use
+   * @param queryString the query string
+   * @param print True to output results to stdout. False to log results.
+   * @throws Exception
+   */
+  public static void testRunAndOutput(final DrillClient drillClient,
+                                      final String queryString,
+                                      final boolean print) throws Exception{
+    final String query = normalizeQuery(queryString);
```

Comment:
Looks query extraction logic is similar here and in `testRunAndLog`, `testRunAndPrint`, can be extracted in private method.

Reply:
Fixed.